tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion.cc

```diff
@@ -392,13 +392,17 @@ Status HorizontalLoopFusionImpl::CreateFusedComputation(
       const HloInstruction* old_output =
           GetOutputsOfFusible(*fused_fusion_instrs[j])[i];
       HloInstruction* new_output = clone_map[old_output];
-      TF_ASSIGN_OR_RETURN(
-          reshapes[j],
-          MakeReshapeHlo(ShapeUtil::MakeShapeWithLayout(
+      Shape new_shape = ShapeUtil::MakeShapeWithLayout(
                              new_output->shape().element_type(),
                              {ShapeUtil::ElementsIn(new_output->shape())},
-                             /*minor_to_major=*/std::vector<int64_t>(1, 0)),
-                         new_output));
+                             /*minor_to_major=*/std::vector<int64_t>(1, 0));
+      if (ShapeUtil::Equal(new_shape, new_output->shape())) {
+        reshapes[j] = new_output;
+      } else {
+        TF_ASSIGN_OR_RETURN(
+            reshapes[j],
+            MakeReshapeHlo(new_shape, new_output));
+      }
```

Comment:
I understand this change does not insert a reshape unless necessary, is this the "main" change associated with the PR? If yes, could it be reflected in PR description?

Reply:
I updated the description with the 3 main changes. This is one of them.